### DISCUSSION TITLE
SSO Grafana and my WebApp

### DISCUSSION THREAD

**Discussion by edbayou** [dexidp/dex#3000] [Q&A]
how can i go from my site on vuejs to grafana, 
logging in to dex once? 
i configured vue app to oidc, and grafana too. 
but when I go to grafana, it asks for the password again.
dex config:
oauth2:
      responseTypes: ["code", "token", "id_token"]
      skipApprovalScreen: true
      alwaysShowLoginScreen: false
    staticClients:
    - id: grafana_client_id
      name: 'Grafana'
      secret: grafana_client_secret
      redirectURIs:
        - 'https://domain/grafana/login/generic_oauth'
    - id: web_client_id
      name: 'Web'
      secret: web_client_secret
      redirectURIs:
        - 'https://domain/signin-oidc'
        - 'https://domain/callback'

grafana conf:
  GF_AUTH_GENERIC_OAUTH_CLIENT_ID: "grafana_client_id"
  GF_AUTH_GENERIC_OAUTH_CLIENT_SECRET: "grafana_client_secret"
  GF_AUTH_GENERIC_OAUTH_ALLOW_SIGN_UP: "true"
  GF_AUTH_GENERIC_OAUTH_API_URL: https://domain/dex/userinfo
  GF_AUTH_GENERIC_OAUTH_AUTH_URL: https://domain/dex/auth
  GF_AUTH_GENERIC_OAUTH_ENABLED: "true"
  GF_AUTH_GENERIC_OAUTH_NAME: Dex
  GF_AUTH_GENERIC_OAUTH_ROLE_ATTRIBUTE_PATH: contains(groups[*], 'admins') && 'Admin'  || contains(groups[*], 'editors') && 'Editor' || 'Viewer'
  GF_AUTH_GENERIC_OAUTH_SCOPES: openid groups email profile
  GF_AUTH_GENERIC_OAUTH_TLS_SKIP_VERIFY_INSECURE: "true"
  GF_AUTH_GENERIC_OAUTH_TOKEN_URL: https://domain/dex/token
  GF_AUTH_OAUTH_AUTO_LOGIN: "true"

config oidc client on web app:
const settings: any = {
            userStore: new WebStorageStateStore({ store: window.localStorage }),
            authority: "https://domain/dex"
            client_id: 'web_client_id',
            client_secret: 'web_client_secret',
            redirect_uri: config.BASE_URL + "/callback",
            silent_redirect_uri: config.BASE_URL + '/silent-renew',
            response_type: 'code',
            response_mode: 'query',
            scope: 'openid email groups profile',
            post_logout_redirect_uri: config.BASE_URL ,
            automaticSilentRenew: true,
            loadUserInfo: true
}
how to make the transition from a web application to Grafana?

**Comment by maonie** [comment]
Have you solved you problem? I am encountering the same issue.
